auto_submit/test/service/validation_service_test.dart

```diff
@@ -268,6 +269,183 @@ void main() {
       // We acknowledge the issue.
       assert(pubsub.messagesQueue.isEmpty);
     });
+
+    test('Exhaust retries on merge on retryable error.', () async {
+      validationService = ValidationService(
+        config,
+        retryOptions:
+            const RetryOptions(delayFactor: Duration(milliseconds: 50), maxDelay: Duration(seconds: 1), maxAttempts: 3),
```

Comment:
Avoid delays in test: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#never-check-if-a-port-is-available-before-using-it-never-add-timeouts-and-other-race-conditions

Reply:
According to that it is talking about sleeping until something occurs. 
> avoid delays or sleeps that are intended to coincide with how long something takes
I am not doing this. I am letting the function run a set amount of time (it will always terminate and run to complete) rather than trying to 'catch' a condition by attempting to sleep until it happens then looking for that condition in the test.

Reply:
Such delays in test do not help but increase the overall test execution time. In prod, we need the delay to wait for Github process to finish. But in the test, we are mocking behaviors which do not depend on any delays.

Reply:
Okay. No problem. I can implement it without the delay.